Allow Job.Service to access the underlying Toil Job (resolves #1191)

fnothaft opened the pull request:
Changes the `Job.Service` `start` and `stop` methods to pass `job` instead of `fileStore`, which allows `Job.Service` instances to register defer methods, etc. To do this, we also modify `ServiceJob.run` to set `self.fileStore` within the service job, which was previously not exposed.

hannes-ucsc commented on Makefile:
```diff
@@ -92,9 +92,8 @@ clean_sdist:
 	- rm -rf dist
 
 
-test: check_venv check_build_reqs docker
```
This shouldn't be committed.

fnothaft replied:
Yes, sorry, did this for local testing. Fixed.

hannes-ucsc commented on src/toil/job.py:
```diff
@@ -1384,12 +1385,16 @@ def __init__(self, service):
     def run(self, fileStore):
+        
+        # we need access to the filestore from underneath the service job
+        self.fileStore = fileStore
```
The fact that you have to do this is a bad smell, coming from Toil. JobFunctionWrappingJob does the same assignment but other Job subclasses don't, most likely because they don't need it. We should decide if we pass the fileStore down or if we set it on the job *before* `run()` is being invoked.

hannes-ucsc replied:
I think it should be pretty easy to make sure that every job has `.fileStore` but if you don't want to do this as part of this PR, could you create a ticket?

fnothaft commented:
@hannes-ucsc tests are passing, ping for a review. If it looks good to you, I can squash. We need this for https://github.com/BD2KGenomics/toil-lib/pull/46 downstream.

fnothaft commented:
> I think it should be pretty easy to make sure that every job has .fileStore but if you don't want to do this as part of this PR, could you create a ticket?

Agreed. It is more of a bandwidth issue, unfortunately. I have a deadline next Thursday and thus don't have time for non-critical path issues. Since this is blocking @almussel via the upstream PR in toil-lib, I want to get this in. I've just squashed and rebased.

fnothaft commented:
Opened https://github.com/BD2KGenomics/toil/issues/1224, will follow up later.

fnothaft commented:
Jenkins, test this please.

fnothaft commented:
Jenkins, test this please.

fnothaft commented:
Can I get a merge?

hannes-ucsc commented:
The prospects of a linear history are starting to grow on me. If only I could get Jenkins to build a rebase instead of a merge. I do like the fact that the commit that's built by the PR is identical to the one that lands on master.
